Source/Mothership/Mothership.cpp

```diff
@@ -0,0 +1,1420 @@
+//------------------------------------------------------------------------------
+
+#include "Mothership.h"
+#include <stdio.h>
+#include <sstream>
+#include <dlfcn.h>
+#include "Pglobals.h"
+#include "CMsg_p.h"
+#include "flat.h"
+#include "string.h"
+#include "limits.h"
+
+const int   Mothership::NumBoards;
+
+//==============================================================================
+
+Mothership::Mothership(int argc,char * argv[],string d) :
+  SBase(argc,argv,d,string(__FILE__)), HostLink()
+{
+
+
+FnMapx.push_back(new FnMap_t);    // create a new event map in the derived class
+
+// Load the incoming event map

```

Comment:
You can't just replace it with Connect, although it's possible to replace the section setting up the function maps for each comm into a private function, but if this is done, I'd rather make that the subject of a separate set of modifications and PR because the same logic applies in all processes. Better to do them all in one go than to modify just Mothership.